--- dvdauthor-0.6.14/src/readxml.c~	2007-01-11 02:43:37.000000000 +0100
+++ dvdauthor-0.6.14/src/readxml.c	2025-08-20 08:41:15.923582232 +0200
@@ -30,6 +30,10 @@
 
 #include "readxml.h"
 
+#ifdef HAVE_ICONV
+#include <iconv.h>
+#endif
+
 #ifdef HAVE_LANGINFO_CODESET
 #include <langinfo.h>
 #include <locale.h>
--- dvdauthor-0.6.14/src/subgen-image.c~	2007-01-13 02:40:42.000000000 +0100
+++ dvdauthor-0.6.14/src/subgen-image.c	2025-08-20 08:44:45.608172756 +0200
@@ -203,7 +203,7 @@ static int read_png(pict *s)
     png_struct *ps;
     png_info *pi;
     png_byte **rowp;
-    unsigned long width,height;
+    png_uint_32 width,height;
     int bit_depth,color_type,channels,x,y;
 
     fp=fopen(s->fname,"rb");
